libkbfs/disk_journal.go

```diff
@@ -146,17 +149,16 @@ func (j diskJournal) writeLatestOrdinal(o journalOrdinal) error {
 	return j.writeOrdinal(j.latestPath(), o)
 }
 
-func (j diskJournal) clearOrdinals() error {
-	earliestOrdinal, err := j.readEarliestOrdinal()
-	if err != nil {
-		return err
-	}
-	latestOrdinal, err := j.readLatestOrdinal()
-	if err != nil {
-		return err
-	}
-
-	err = ioutil.Remove(j.earliestPath())
+// clear completely removes the journal directory.
+func (j diskJournal) clear() error {
+	// Clear ordinals first to reduce the chances of leaving the
+	// journal in a weird state if we crash in the middle of
+	// removing the files.
+	//
+	// TODO: When we read ordinals into memory on startup, treat
```

Comment:
Is the idea that, as part of this TODO, it would call `clear()` again before proceeding?  Would it be possible to just do that as part of this PR?  Otherwise it introduces a new failure mode.  Doesn't seem like it would be hard to add, but let me know if I'm wrong.